cmd/dlt/ocmrole/cmd.go

```diff
@@ -0,0 +1,239 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+package ocmrole
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/spf13/cobra"
+
+	unlinkocmrole "github.com/openshift/rosa/cmd/unlink/ocmrole"
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/helper"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/logging"
+	"github.com/openshift/rosa/pkg/ocm"
+	rprtr "github.com/openshift/rosa/pkg/reporter"
+)
+
+var args struct {
+	roleARN string
+}
+
+var Cmd = &cobra.Command{
+	Use:     "ocm-role",
+	Aliases: []string{"ocmrole"},
+	Short:   "Delete ocm role.",
+	Long:    "Delete ocm role from the current AWS account.",
+	Example: ` # Delete ocm role
+rosa delete ocm-role --role-arn arn:aws:iam::123456789012:role/xxx-OCM-Role-1223456778`,
+	RunE: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVar(
+		&args.roleARN,
+		"role-arn",
+		"",
+		"Role ARN to delete from the OCM organization account")
+
+	aws.AddModeFlag(Cmd)
+
+	confirm.AddFlag(flags)
+	interactive.AddFlag(flags)
+}
+
+func run(cmd *cobra.Command, argv []string) error {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+	awsClient := aws.CreateNewClientOrExit(logger, reporter)
+	ocmClient := ocm.CreateNewClientOrExit(logger, reporter)
+	defer func() {
+		err := ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	mode, err := aws.GetMode()
+	if err != nil {
+		reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
+	orgID, _, err := ocmClient.GetCurrentOrganization()
+	if err != nil {
+		reporter.Errorf("Error getting organization account: %v", err)
+		return err
+	}
+
+	if len(argv) > 0 {
+		args.roleARN = argv[0]
+	}
+
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && (!cmd.Flags().Changed("mode")) {
+		interactive.Enable()
+	}
+
+	if reporter.IsTerminal() {
+		reporter.Infof("Deleting OCM role")
+	}
+
+	roleARN := args.roleARN
+
+	if !interactive.Enabled() && roleARN == "" {
+		interactive.Enable()
+	}
+
+	if interactive.Enabled() {
+		roleARN, err = interactive.GetString(interactive.Input{
+			Question: "OCM Role ARN",
+			Help:     cmd.Flags().Lookup("role-arn").Usage,
+			Default:  roleARN,
+			Required: true,
+			Validators: []interactive.Validator{
+				aws.ARNValidator,
+			},
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid ocm role ARN to delete from the current organization: %s", err)
+			os.Exit(1)
+		}
+	}
+	if roleARN != "" {
+		_, err := arn.Parse(roleARN)
+		if err != nil {
+			reporter.Errorf("Expected a valid ocm role ARN to delete from the current organization: %s", err)
+			os.Exit(1)
+		}
+	}
+	if !confirm.Prompt(true, "Delete the '%s' role from organization '%s'?", roleARN, orgID) {
```

Comment:
We dont need to specify organization.  Delete  '%s' ocm role ?

Reply:
Changed.